scipy/spatial/transform/_rotation.pyx

```diff
@@ -2515,33 +2515,31 @@ cdef class Rotation:
             raise ValueError("Expected input of shape (3,) or (P, 3), "
                              "got {}.".format(vectors.shape))
 
-        single_vector = False
-        if vectors.shape == (3,):
-            single_vector = True
-            vectors = vectors[None, :]
-
         matrix = self.as_matrix()
+        if inverse:
+            matrix = np.swapaxes(matrix, -1, -2)
+
+        if vectors.shape == (3,):
```

Comment:
Is it possible to move it somehow into the main computation block?

Will it work when `matrix` has 3 dimensions?